src/Chapter1.hs

```diff
@@ -635,8 +650,12 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n = firstDigit' (abs n)
+    where
+        firstDigit' absN
+            | absN < 10 = absN
+            | otherwise = firstDigit' (div absN 10)
```

Comment:
Regarding your questions about idioms on recursive functions. Usually, a helper recursive function in Haskell is called `go`. It is described in our blog post about Haskell patterns:

* https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

We explain recursion in more details in the next Chapter :slightly_smiling_face: